### DISCUSSION TITLE
What is the reason for subsequent root navigation not triggering hooks handle function?

### DISCUSSION THREAD

**Discussion by laszlo1337** [sveltejs/kit#10945] [Q&A] [answered]
I couldn't find anything about it in the docs, but I cannot understand that as AFAIK all requests should go thru hooks.server.js if defined.

so just to clarify, the flow:
1. go to http://localhost:5173/ for the first time (by entering address in address bar)
2. the handle function runs normally
3. navigate away to different route eg /shop using html anchor
4. handle function runs for /shop
5. navigate back to root / using html anchor
6. handle function for / **doesn't run**
7. if navigated to root once again by browser address then the function runs normally

I don't plan for root path to be authed, but if somebody wanted that then it will be a problem. I just stumbled upon this by accident.

EDIT: 
sveltekit version is 1.27.1

EDIT2:

I think it doesn't run if root page doesn't have +page.server.js file. Just tested that and it appears that's how it works. Anybody can point me to the docs where that is mentioned?

**Comment by teemingc** [comment, marked as answer]
As far as I recall, it's not explicitly mentioned anywhere in the documentation. I'd highly encourage you to submit an issue for this.

The reasoning for your handle function running depends on how your project setup. Assuming the page options are left as default (`ssr` and `csr` enabled), the initial request will server-side render the page (goes through the server hook `handle`). Subsequent requests are client-side rendered, so only if they have a server load function, a request will be sent to request the `__data.json` file, which goes through the `handle` hook once again. Otherwise, they never trigger the server hook `handle`.

The link below is an example of the server hook `handle` logging any request that comes through. Notice that it only logs the initial request to the root page. Navigating to the `/about` page for the first time does not trigger the hook as there is no server load function (so no `__data.json` to request).
https://stackblitz.com/edit/sveltejs-kit-template-default-ocdjxb?file=src%2Fhooks.server.js

**Reply by Neptunium1129** [reply]
no one is interested in this issue..?

**Reply by Neptunium1129** [reply]
This issue occurred in SvelteKit 2. I opened two tabs with the same URL. After logging out in tab B, tab A doesn't automatically redirect to the login page. The session is only revalidated in hooks.server.ts when I refresh the page or re-enter the URL.

**Reply by teemingc** [reply]
Can you submit a new issue with a minimal reproduction?

**Reply by orenaksakal** [reply]
I just figured out; the path you want hooks.server.ts to be executed on must have +page.server.ts even if its empty 🤯

**Reply by teemingc** [reply]
> I just figured out; the path you want hooks.server.ts to be executed on must have +page.server.ts even if its empty 🤯

Yes, if your route does not invoke the server (no server load function or does not need to be SSR-ed), then the server handle hook will not be triggered.
